smartcontractkit/external-adapters-js
# Fix integration tests WASM OOM error

## Description

Integration tests fail both locally and remotely with WebAssembly.Instance(): Out of memory: wasm memory. It seems to be due to a linear increase in heap usage. As a fix, tests will be modified to run in parallel.

## Changes

- Integration tests now run with port `0`, thereby using random available ports provided by the system.
- Integration tests run in parallel, to mitigate heap memory issues by splitting load into many jest workers.
- Adds types for Supertest to many missing adapter tests



## Steps to Test

1. Run `yarn test:integration` and verify integration tests run in parallel, all passing
2. Run `yarn test:ci:integration` and verify integration tests run in parallel, all passing

## Quality Assurance
